addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -32,12 +32,12 @@ def _default_country_ids(self):
         return self.env.user.company_id.country_id
 
     name = fields.Char(string='Request Number', required=True, readonly=True, default=lambda self: _('New'), copy=False)
-    state = fields.Selection([('draft', 'Draft'), ('done', 'Done'), ('error', 'Error')], string='Status', required=True, default='draft')
+    state = fields.Selection([('draft', 'Draft'), ('error', 'Error'), ('done', 'Done')], string='Status', required=True, default='draft')
 
     # Request Data
     lead_number = fields.Integer(string='Number of Leads', required=True, default=3)
     search_type = fields.Selection([('companies', 'Companies'), ('people', 'Companies and their Contacts')], string='Target', required=True, default='companies')
-    error = fields.Text(string='Error', readonly=True)
+    error_code = fields.Char(string='Error Code', readonly=True)
```

Comment:
shouldn't it be a selection field instead ? As we have only 2 possible values here ?

Reply:
+1 for selection field (with minimal keys if possible)